File: fungal-cortex/skills/openclaw-master-skills-main/openclaw-master-skills-main/skills/fundraising-advisor/fa_advisor/modules/assessment/projectAssessor.py
"""
Project Assessor - Evaluate startup projects for investment readiness
"""

import logging
from typing import List

from fa_advisor.types.project import Project
from fa_advisor.types.models import (
    ProjectAssessment,
    AssessmentScores,
    InvestmentReadiness
)

logger = logging.getLogger(__name__)


class ProjectAssessor:
    """
    Assess startup projects across multiple dimensions

    Scoring dimensions:
    - Team (founders, size, experience)
    - Market (TAM, growth, competition)
    - Product (stage, features, differentiation)
    - Traction (customers, revenue, growth)
    - Financials (metrics, runway, unit economics)
    """

    async def assess(self, project: Project) -> ProjectAssessment:
        """
        Comprehensive project assessment

        Returns:
            ProjectAssessment with scores, readiness, and recommendations
        """
        logger.info(f"Assessing project: {project.name}")

        # Calculate individual dimension scores
        team_score = self._assess_team(project)
        market_score = self._assess_market(project)
        product_score = self._assess_product(project)
        traction_score = self._assess_traction(project)
        financials_score = self._assess_financials(project)

        # Calculate overall score
        overall_score = (
            team_score + market_score + product_score +
            traction_score + financials_score
        ) / 5

        scores = AssessmentScores(
            team=team_score,
            market=market_score,
            product=product_score,
            traction=traction_score,
            financials=financials_score,
            overall=overall_score
        )

        # Determine investment readiness
        readiness = self._determine_readiness(overall_score)

        # Identify strengths and weaknesses
        strengths = self._identify_strengths(project, scores)
        weaknesses = self._identify_weaknesses(project, scores)

        # Generate recommendations
        recommendations = self._generate_recommendations(project, scores, weaknesses)

        # Generate summary
        summary = self._generate_summary(project, scores, readiness)

        return ProjectAssessment(
            scores=scores,
            investment_readiness=readiness,
            strengths=strengths,
            weaknesses=weaknesses,
            recommendations=recommendations,
            summary=summary
        )

    def _assess_team(self, project: Project) -> float:
        """Assess team quality (0-100)"""
        score = 0.0

        # Founder quality (40 points)
        founders = project.team.founders
        if len(founders) >= 2:
            score += 15  # Multiple founders
        elif len(founders) == 1:
            score += 8

        # Check for relevant experience in founder backgrounds
        for founder in founders:
            if any(keyword in founder.background.lower()
                   for keyword in ['founder', 'ceo', 'cto', 'vp', 'director']):
                score += 10
                break

        # LinkedIn presence
        if any(f.linkedin for f in founders):
            score += 5

        # Team size (30 points)
        team_size = project.team.team_size
        if team_size >= 20:
            score += 30
        elif team_size >= 10:
            score += 25
        elif team_size >= 5:
            score += 20
        elif team_size >= 3:
            score += 15
        else:
            score += 10

        # Key hires (15 points)
        if project.team.key_hires and len(project.team.key_hires) > 0:
            score += 15

        # Balance check (15 points)
        if len(founders) >= 2 and team_size >= 5:
            score += 15  # Good founder-team balance

        return min(score, 100)

    def _assess_market(self, project: Project) -> float:
        """Assess market opportunity (0-100)"""
        score = 0.0
        market = project.market

        # TAM size (40 points)
        tam = market.tam
        if tam >= 50_000_000_000:  # $50B+
            score += 40
        elif tam >= 10_000_000_000:  # $10B+
            score += 35
        elif tam >= 1_000_000_000:  # $1B+
            score += 30
        elif tam >= 100_000_000:  # $100M+
            score += 20
        else:
            score += 10

        # Growth rate (30 points)
        growth_rate = market.market_growth_rate
        if growth_rate >= 0.4:  # 40%+
            score += 30
        elif growth_rate >= 0.25:  # 25%+
            score += 25
        elif growth_rate >= 0.15:  # 15%+
            score += 20
        elif growth_rate >= 0.05:  # 5%+
            score += 10
        else:
            score += 5

        # Competition analysis (20 points)
        competitors = market.competitors
        if len(competitors) == 0:
            score += 10  # Blue ocean (but maybe risky)
        elif 1 <= len(competitors) <= 3:
            score += 20  # Good competitive position
        elif len(competitors) <= 5:
            score += 15  # Competitive but manageable
        else:
            score += 5  # Highly competitive

        # SAM/SOM provided (10 points)
        if market.sam and market.som:
            score += 10  # Detailed market analysis

        return min(score, 100)

    def _assess_product(self, project: Project) -> float:
        """Assess product maturity (0-100)"""
        score = 0.0
        product = project.product

        # Product stage (40 points)
        stage_scores = {
            'idea': 10,
            'mvp': 25,
            'launched': 35,
            'scaling': 40
        }
        score += stage_scores.get(product.stage.value, 10)

        # Key features (20 points)
        num_features = len(product.key_features)
        if num_features >= 5:
            score += 20
        elif num_features >= 3:
            score += 15
        elif num_features >= 1:
            score += 10

        # Clear value proposition (20 points)
        uvp_length = len(product.unique_value_proposition)
        if uvp_length >= 50:
            score += 20
        elif uvp_length >= 20:
            score += 15
        else:
            score += 5

        # Pain points identified (20 points)
        pain_points = product.customer_pain_points
        if len(pain_points) >= 3:
            score += 20
        elif len(pain_points) >= 2:
            score += 15
        elif len(pain_points) >= 1:
            score += 10

        return min(score, 100)

    def _assess_traction(self, project: Project) -> float:
        """Assess market traction (0-100)"""
        score = 0.0
        traction = project.traction
        revenue = project.financials.revenue.current

        # Revenue (50 points)
        if revenue >= 5_000_000:  # $5M+
            score += 50
        elif revenue >= 1_000_000:  # $1M+
            score += 45
        elif revenue >= 500_000:  # $500K+
            score += 40
        elif revenue >= 100_000:  # $100K+
            score += 30
        elif revenue > 0:
            score += 20
        else:
            score += 10  # Pre-revenue

        # Customers (20 points)
        if traction and traction.customers:
            customers = traction.customers
            if customers >= 100:
                score += 20
            elif customers >= 50:
                score += 18
            elif customers >= 20:
                score += 15
            elif customers >= 10:
                score += 12
            elif customers >= 1:
                score += 8

        # Partnerships (15 points)
        if traction and traction.partnerships and len(traction.partnerships) > 0:
            score += 15

        # Press/Awards (15 points)
        media_count = 0
        if traction:
            if traction.awards:
                media_count += len(traction.awards)
            if traction.press:
                media_count += len(traction.press)

        if media_count >= 5:
            score += 15
        elif media_count >= 2:
            score += 10
        elif media_count >= 1:
            score += 5

        return min(score, 100)

    def _assess_financials(self, project: Project) -> float:
        """Assess financial health (0-100)"""
        score = 0.0
        financials = project.financials

        # Runway (30 points)
        runway = financials.expenses.runway
        if runway >= 18:
            score += 30
        elif runway >= 12:
            score += 25
        elif runway >= 6:
            score += 15
        elif runway >= 3:
            score += 5

        # Metrics available (30 points)
        metrics = financials.metrics
        if metrics:
            # Gross margin
            if metrics.gross_margin:
                if metrics.gross_margin >= 0.8:
                    score += 10
                elif metrics.gross_margin >= 0.6:
                    score += 8
                elif metrics.gross_margin >= 0.4:
                    score += 5

            # LTV:CAC ratio
            if metrics.lifetime_value and metrics.customer_acquisition_cost:
                ltv_cac = metrics.lifetime_value / metrics.customer_acquisition_cost
                if ltv_cac >= 3:
                    score += 10
                elif ltv_cac >= 2:
                    score += 7
                elif ltv_cac >= 1:
                    score += 3

            # Churn rate
            if metrics.churn_rate is not None:
                if metrics.churn_rate <= 0.03:
                    score += 10
                elif metrics.churn_rate <= 0.05:
                    score += 7
                elif metrics.churn_rate <= 0.10:
                    score += 3

        # Revenue projections (20 points)
        projections = financials.revenue.projected
        if len(projections) >= 3:
            score += 15
            # Check growth rate
            if len(projections) >= 2:
                growth = (projections[1].amount - projections[0].amount) / projections[0].amount
                if growth >= 2.0:  # 200%+ growth
                    score += 5
        elif len(projections) >= 1:
            score += 10

        # Funding efficiency (20 points)
        monthly_burn = financials.expenses.monthly
        revenue_current = financials.revenue.current
        if revenue_current > monthly_burn:
            score += 20  # Default profitability!
        elif monthly_burn > 0:
            burn_multiple = revenue_current / monthly_burn
            if burn_multiple >= 0.8:
                score += 15
            elif burn_multiple >= 0.5:
                score += 10
            elif burn_multiple >= 0.3:
                score += 5

        return min(score, 100)

    def _determine_readiness(self, overall_score: float) -> InvestmentReadiness:
        """Determine investment readiness level"""
        if overall_score >= 80:
            return InvestmentReadiness.HIGHLY_READY
        elif overall_score >= 65:
            return InvestmentReadiness.READY
        elif overall_score >= 50:
            return InvestmentReadiness.NEEDS_IMPROVEMENT
        else:
            return InvestmentReadiness.NOT_READY

    def _identify_strengths(self, project: Project, scores: AssessmentScores) -> List[str]:
        """Identify project strengths"""
        strengths = []

        if scores.team >= 75:
            strengths.append("Strong, experienced founding team")
        if scores.market >= 75:
            strengths.append("Large and rapidly growing market opportunity")
        if scores.product >= 75:
            strengths.append("Mature product with clear differentiation")
        if scores.traction >= 75:
            strengths.append("Solid market traction and customer validation")
        if scores.financials >= 75:
            strengths.append("Healthy financial metrics and runway")

        # Specific strengths
        if project.financials.revenue.current > 1_000_000:
            strengths.append(f"Significant revenue: ${project.financials.revenue.current:,.0f}")
        if project.market.tam >= 10_000_000_000:
            strengths.append(f"Massive TAM: ${project.market.tam / 1_000_000_000:.1f}B")
        if project.team.team_size >= 20:
            strengths.append(f"Substantial team size: {project.team.team_size} people")

        return strengths[:5]  # Top 5 strengths

    def _identify_weaknesses(self, project: Project, scores: AssessmentScores) -> List[str]:
        """Identify project weaknesses"""
        weaknesses = []

        if scores.team < 50:
            weaknesses.append("Team needs strengthening - consider key hires")
        if scores.market < 50:
            weaknesses.append("Market opportunity unclear or too small")
        if scores.product < 50:
            weaknesses.append("Product needs more development and differentiation")
        if scores.traction < 50:
            weaknesses.append("Limited market traction and customer validation")
        if scores.financials < 50:
            weaknesses.append("Financial metrics need improvement")

        # Specific weaknesses
        if project.financials.expenses.runway < 6:
            weaknesses.append(f"Short runway: only {project.financials.expenses.runway} months")
        if project.financials.revenue.current == 0:
            weaknesses.append("Pre-revenue - need customer validation")
        if len(project.market.competitors) > 5:
            weaknesses.append("Highly competitive market")

        return weaknesses[:5]  # Top 5 weaknesses

    def _generate_recommendations(
        self,
        project: Project,
        scores: AssessmentScores,
        weaknesses: List[str]
    ) -> List[str]:
        """Generate actionable recommendations"""
        recommendations = []

        if scores.team < 70:
            recommendations.append("Strengthen team with key executive hires (VP Sales, Head of Product)")
        if scores.market < 70 and project.market.tam < 1_000_000_000:
            recommendations.append("Expand TAM analysis or pivot to larger market opportunity")
        if scores.product < 70:
            recommendations.append("Accelerate product development and gather more customer feedback")
        if scores.traction < 70 and project.financials.revenue.current < 500_000:
            recommendations.append("Focus on customer acquisition and revenue growth")
        if scores.financials < 70:
            if project.financials.expenses.runway < 12:
                recommendations.append("Extend runway through cost optimization or bridge funding")
            if project.financials.metrics and project.financials.metrics.churn_rate:
                if project.financials.metrics.churn_rate > 0.05:
                    recommendations.append("Improve customer retention - churn rate is high")

        # Stage-specific recommendations
        if project.fundraising.current_stage.value in ['pre-seed', 'seed']:
            recommendations.append("Build MVP and achieve product-market fit before scaling")

        return recommendations[:5]  # Top 5 recommendations

    def _generate_summary(
        self,
        project: Project,
        scores: AssessmentScores,
        readiness: InvestmentReadiness
    ) -> str:
        """Generate assessment summary"""
        stage = project.fundraising.current_stage.value.title()
        score = scores.overall

        summary = f"{project.name} is a {stage}-stage {project.industry.value} company "
        summary += f"with an overall assessment score of {score:.0f}/100, "
        summary += f"indicating {readiness.value} for fundraising. "

        # Highlight top dimension
        top_dimension = max(
            [('Team', scores.team), ('Market', scores.market),
             ('Product', scores.product), ('Traction', scores.traction),
             ('Financials', scores.financials)],
            key=lambda x: x[1]
        )
        summary += f"The company's strongest dimension is {top_dimension[0]} ({top_dimension[1]:.0f}/100)."

        return summary

File: fungal-cortex/skills/openclaw-master-skills-main/openclaw-master-skills-main/skills/fundraising-advisor/src/modules/assessment/projectAssessor.ts
import type { Project, ProjectAssessment } from '../../types/project.js';

/**
 * 项目评估引擎
 * 全面评估创业项目的投资价值
 */
export class ProjectAssessor {
  /**
   * 评估项目
   */
  async assess(project: Project): Promise<ProjectAssessment> {
    const scores = {
      team: this.assessTeam(project),
      market: this.assessMarket(project),
      product: this.assessProduct(project),
      traction: this.assessTraction(project),
      financials: this.assessFinancials(project),
      overall: 0,
    };

    // 加权计算总分
    scores.overall = this.calculateOverallScore(scores);

    const strengths = this.identifyStrengths(project, scores);
    const weaknesses = this.identifyWeaknesses(project, scores);
    const risks = this.assessRisks(project);
    const recommendations = this.generateRecommendations(project, scores, weaknesses);
    const investmentReadiness = this.determineInvestmentReadiness(scores, risks);

    return {
      project,
      scores,
      strengths,
      weaknesses,
      risks,
      recommendations,
      investmentReadiness,
    };
  }

  /**
   * 评估团队 (0-100)
   */
  private assessTeam(project: Project): number {
    let score = 0;
    const { team } = project;

    // 创始人数量 (0-20)
    const founderCount = team.founders.length;
    if (founderCount >= 2 && founderCount <= 4) {
      score += 20;
    } else if (founderCount === 1) {
      score += 10;
    } else {
      score += 5;
    }

    // 创始人背景 (0-30)
    const hasRelevantBackground = team.founders.some(
      f => f.background.toLowerCase().includes('experience') ||
           f.background.toLowerCase().includes('expertise')
    );
    if (hasRelevantBackground) {
      score += 30;
    } else {
      score += 15;
    }

    // 团队规模 (0-20)
    if (team.teamSize >= 5 && team.teamSize <= 50) {
      score += 20;
    } else if (team.teamSize >= 2 && team.teamSize < 5) {
      score += 15;
    } else if (team.teamSize > 50) {
      score += 10;
    } else {
      score += 5;
    }

    // 关键岗位 (0-30)
    const hasKeyHires = team.keyHires && team.keyHires.length > 0;
    if (hasKeyHires) {
      score += 30;
    } else {
      score += 10;
    }

    return Math.min(score, 100);
  }

  /**
   * 评估市场 (0-100)
   */
  private assessMarket(project: Project): number {
    let score = 0;
    const { market } = project;

    // 市场规模 (0-40)
    if (market.tam !== undefined) {
      if (market.tam >= 10_000_000_000) { // $10B+
        score += 40;
      } else if (market.tam >= 1_000_000_000) { // $1B+
        score += 30;
      } else if (market.tam >= 100_000_000) { // $100M+
        score += 20;
      } else {
        score += 10;
      }
    }

    // 市场增长率 (0-30)
    if (market.marketGrowthRate !== undefined) {
      if (market.marketGrowthRate >= 0.3) { // 30%+
        score += 30;
      } else if (market.marketGrowthRate >= 0.15) { // 15%+
        score += 20;
      } else if (market.marketGrowthRate >= 0.05) { // 5%+
        score += 10;
      } else {
        score += 5;
      }
    }

    // 竞争格局 (0-30)
    const competitorCount = market.competitors.length;
    const hasDifferentiation = market.competitors.some(c => c.differentiation);

    if (competitorCount <= 3 && hasDifferentiation) {
      score += 30;
    } else if (competitorCount <= 5) {
      score += 20;
    } else if (competitorCount <= 10) {
      score += 10;
    } else {
      score += 5;
    }

    return Math.min(score, 100);
  }

  /**
   * 评估产品 (0-100)
   */
  private assessProduct(project: Project): number {
    let score = 0;
    const { product } = project;

    // 产品阶段 (0-40)
    const stageScores: Record<typeof product.stage, number> = {
      'idea': 5,
      'prototype': 15,
      'mvp': 25,
      'launched': 35,
      'scaling': 40,
    };
    score += stageScores[product.stage];

    // 独特价值主张 (0-30)
    const uvpLength = product.uniqueValueProposition.length;
    if (uvpLength >= 50) {
      score += 30;
    } else if (uvpLength >= 20) {
      score += 20;
    } else {
      score += 10;
    }

    // 核心功能 (0-15)
    const featureCount = product.keyFeatures.length;
    if (featureCount >= 5) {
      score += 15;
    } else if (featureCount >= 3) {
      score += 10;
    } else {
      score += 5;
    }

    // 痛点识别 (0-15)
    const painPointCount = product.customerPainPoints.length;
    if (painPointCount >= 3) {
      score += 15;
    } else if (painPointCount >= 2) {
      score += 10;
    } else {
      score += 5;
    }

    return Math.min(score, 100);
  }

  /**
   * 评估牵引力 (0-100)
   */
  private assessTraction(project: Project): number {
    if (!project.traction) {
      return 0;
    }

    let score = 0;
    const { traction } = project;

    // 客户数量 (0-30)
    if (traction.customers !== undefined) {
      if (traction.customers >= 1000) {
        score += 30;
      } else if (traction.customers >= 100) {
        score += 20;
      } else if (traction.customers >= 10) {
        score += 10;
      } else if (traction.customers > 0) {
        score += 5;
      }
    }

    // 用户数量 (0-25)
    if (traction.users !== undefined) {
      if (traction.users >= 100_000) {
        score += 25;
      } else if (traction.users >= 10_000) {
        score += 20;
      } else if (traction.users >= 1_000) {
        score += 15;
      } else if (traction.users > 0) {
        score += 10;
      }
    }

    // 合作伙伴 (0-20)
    if (traction.partnerships && traction.partnerships.length > 0) {
      score += Math.min(traction.partnerships.length * 5, 20);
    }

    // 奖项和媒体 (0-15)
    const hasAwards = traction.awards && traction.awards.length > 0;
    const hasPress = traction.press && traction.press.length > 0;
    if (hasAwards && hasPress) {
      score += 15;
    } else if (hasAwards || hasPress) {
      score += 10;
    }

    // 增长趋势 (0-10)
    if (traction.growth) {
      score += 10;
    }

    return Math.min(score, 100);
  }

  /**
   * 评估财务 (0-100)
   */
  private assessFinancials(project: Project): number {
    let score = 0;
    const { financials } = project;

    // 收入 (0-30)
    if (financials.revenue) {
      if (financials.revenue.current >= 10_000_000) { // $10M+
        score += 30;
      } else if (financials.revenue.current >= 1_000_000) { // $1M+
        score += 25;
      } else if (financials.revenue.current >= 100_000) { // $100K+
        score += 15;
      } else if (financials.revenue.current > 0) {
        score += 10;
      }
    }

    // 关键指标 (0-40)
    if (financials.metrics) {
      const { metrics } = financials;

      // 毛利率
      if (metrics.grossMargin !== undefined) {
        if (metrics.grossMargin >= 0.7) {
          score += 15;
        } else if (metrics.grossMargin >= 0.5) {
          score += 10;
        } else if (metrics.grossMargin >= 0.3) {
          score += 5;
        }
      }

      // LTV/CAC比率
      if (metrics.lifetimeValue && metrics.customerAcquisitionCost) {
        const ratio = metrics.lifetimeValue / metrics.customerAcquisitionCost;
        if (ratio >= 3) {
          score += 15;
        } else if (ratio >= 2) {
          score += 10;
        } else if (ratio >= 1) {
          score += 5;
        }
      }

      // 流失率
      if (metrics.churnRate !== undefined) {
        if (metrics.churnRate <= 0.05) { // 5%以下
          score += 10;
        } else if (metrics.churnRate <= 0.1) { // 10%以下
          score += 5;
        }
      }
    }

    // 跑道 (0-30)
    if (financials.expenses) {
      const { runway } = financials.expenses;
      if (runway >= 18) {
        score += 30;
      } else if (runway >= 12) {
        score += 20;
      } else if (runway >= 6) {
        score += 10;
      } else {
        score += 5;
      }
    }

    return Math.min(score, 100);
  }

  /**
   * 计算综合得分
   */
  private calculateOverallScore(scores: Omit<ProjectAssessment['scores'], 'overall'>): number {
    const weights = {
      team: 0.30,
      market: 0.25,
      product: 0.20,
      traction: 0.15,
      financials: 0.10,
    };

    return Math.round(
      scores.team * weights.team +
      scores.market * weights.market +
      scores.product * weights.product +
      scores.traction * weights.traction +
      scores.financials * weights.financials
    );
  }

  /**
   * 识别优势
   */
  private identifyStrengths(project: Project, scores: ProjectAssessment['scores']): string[] {
    const strengths: string[] = [];

    if (scores.team >= 80) {
      strengths.push('优秀的创始团队，具备深厚的行业背景和执行能力');
    }

    if (scores.market >= 80) {
      strengths.push('巨大的市场机会，市场规模可观且增长迅速');
    }

    if (scores.product >= 80) {
      strengths.push('产品已经较为成熟，具备清晰的价值主张和差异化优势');
    }

    if (scores.traction >= 70) {
      strengths.push('已获得显著的市场牵引力，有明确的增长趋势');
    }

    if (scores.financials >= 70) {
      strengths.push('财务指标健康，收入模式已得到验证');
    }

    // 具体特征分析
    if (project.market.tam && project.market.tam >= 10_000_000_000) {
      strengths.push(`TAM超过100亿美元，市场空间巨大`);
    }

    if (project.financials.metrics?.grossMargin && project.financials.metrics.grossMargin >= 0.7) {
      strengths.push(`毛利率${Math.round(project.financials.metrics.grossMargin * 100)}%，盈利能力强`);
    }

    if (project.traction?.customers && project.traction.customers >= 100) {
      strengths.push(`已获得${project.traction.customers}+付费客户，商业模式得到验证`);
    }

    return strengths;
  }

  /**
   * 识别劣势
   */
  private identifyWeaknesses(project: Project, scores: ProjectAssessment['scores']): string[] {
    const weaknesses: string[] = [];

    if (scores.team < 60) {
      weaknesses.push('团队需要加强，建议补充行业经验丰富的核心成员');
    }

    if (scores.market < 60) {
      weaknesses.push('市场机会有限或竞争过于激烈，需要重新评估目标市场');
    }

    if (scores.product < 60) {
      weaknesses.push('产品成熟度不足，需要加快产品迭代和功能完善');
    }

    if (scores.traction < 40) {
      weaknesses.push('缺乏市场验证，需要尽快获取早期客户并证明产品价值');
    }

    if (scores.financials < 40) {
      weaknesses.push('财务指标较弱，需要改善单位经济模型和现金流管理');
    }

    // 具体问题
    if (project.team.founders.length === 1) {
      weaknesses.push('单一创始人风险较高，建议引入联合创始人');
    }

    if (project.financials.expenses && project.financials.expenses.runway < 6) {
      weaknesses.push(`跑道仅剩${project.financials.expenses.runway}个月，需要尽快完成融资`);
    }

    if (project.market.competitors.length > 10) {
      weaknesses.push('竞争过于激烈，需要强化差异化优势');
    }

    if (project.product.stage === 'idea' || project.product.stage === 'prototype') {
      weaknesses.push('产品尚未推向市场，需要加快MVP开发和市场验证');
    }

    return weaknesses;
  }

  /**
   * 风险评估
   */
  private assessRisks(project: Project): ProjectAssessment['risks'] {
    const risks: ProjectAssessment['risks'] = [];

    // 市场风险
    if (project.market.competitors.length > 10) {
      risks.push({
        category: '市场风险',
        description: '市场竞争激烈，面临众多竞争对手',
        severity: 'high',
        mitigation: '建立清晰的差异化定位，聚焦特定细分市场',
      });
    }

    // 执行风险
    if (project.team.founders.length === 1) {
      risks.push({
        category: '执行风险',
        description: '单一创始人，关键人物风险较高',
        severity: 'medium',
        mitigation: '引入联合创始人或关键高管，建立备份机制',
      });
    }

    // 财务风险
    if (project.financials.expenses && project.financials.expenses.runway < 6) {
      risks.push({
        category: '财务风险',
        description: '现金流跑道不足6个月，面临资金压力',
        severity: 'high',
        mitigation: '立即启动融资流程，同时控制成本延长跑道',
      });
    }

    // 产品风险
    if (project.product.stage === 'idea' || project.product.stage === 'prototype') {
      risks.push({
        category: '产品风险',
        description: '产品尚未推向市场，市场接受度未知',
        severity: 'high',
        mitigation: '尽快完成MVP开发，获取早期用户反馈',
      });
    }

    // 牵引力风险
    if (!project.traction || (project.traction.customers || 0) < 10) {
      risks.push({
        category: '市场验证风险',
        description: '缺乏足够的市场验证和客户牵引力',
        severity: 'medium',
        mitigation: '聚焦获取早期付费客户，建立案例研究',
      });
    }

    return risks;
  }

  /**
   * 生成建议
   */
  private generateRecommendations(
    project: Project,
    scores: ProjectAssessment['scores'],
    weaknesses: string[]
  ): string[] {
    const recommendations: string[] = [];

    // 基于得分的建议
    if (scores.overall >= 80) {
      recommendations.push('项目整体质量优秀，建议立即启动融资流程');
      recommendations.push('可以瞄准顶级VC，争取更好的融资条件');
    } else if (scores.overall >= 60) {
      recommendations.push('项目具备融资基础，建议在改进关键弱项后启动融资');
      recommendations.push('针对性补强团队和产品，提升项目竞争力');
    } else {
      recommendations.push('项目尚不具备融资条件，建议先进行内部优化');
      recommendations.push('聚焦产品开发和早期市场验证，积累更多牵引力');
    }

    // 针对性建议
    if (scores.traction < 50) {
      recommendations.push('优先任务：获取早期客户，证明产品市场契合度');
    }

    if (scores.team < 70) {
      recommendations.push('考虑引入行业专家或经验丰富的联合创始人');
    }

    if (scores.financials < 50 && project.financials.revenue) {
      recommendations.push('改善单位经济模型，关注关键财务指标如LTV/CAC比率');
    }

    // 融资策略建议
    if (project.fundraising.currentStage === 'seed' || project.fundraising.currentStage === 'pre-seed') {
      recommendations.push('对于早期阶段，重点展示团队能力和市场机会');
      recommendations.push('寻找愿意承担早期风险的天使投资人或种子基金');
    } else {
      recommendations.push('对于成长阶段，需要展示清晰的收入增长和单位经济模型');
      recommendations.push('目标机构应该在你的行业有专业认知和资源网络');
    }

    return recommendations;
  }

  /**
   * 判断融资准备度
   */
  private determineInvestmentReadiness(
    scores: ProjectAssessment['scores'],
    risks: ProjectAssessment['risks']
  ): ProjectAssessment['investmentReadiness'] {
    const highRisks = risks.filter(r => r.severity === 'high').length;

    if (scores.overall >= 80 && highRisks === 0) {
      return 'highly-ready';
    } else if (scores.overall >= 65 && highRisks <= 1) {
      return 'ready';
    } else if (scores.overall >= 50) {
      return 'needs-work';
    } else {
      return 'not-ready';
    }
  }
}
